### DISCUSSION TITLE
Akka Streams Flow measuring/tracing

### DISCUSSION THREAD

**Discussion by mirkof** [kamon-io/Kamon#921] [Ideas]
We're using a lot of AkkaStreams in our app. It would be great if we could measure how low does one Flow take. Similar to what Kamon already has for measuring Futures (trace), but for Flow. What's the easiest way to do this?

I mean, there is some kind of [LatenecyTimer](https://github.com/akka/akka-stream-contrib/blob/master/src/main/scala/akka/stream/contrib/latencyTimer.scala) in akka-stream-contrib. Maybe I can reuse that? :thinking:

Or I can also use [Timed](https://github.com/akka/akka-stream-contrib/blob/master/src/main/scala/akka/stream/contrib/Timed.scala) with Histogram just to report the durations. But then, no spans are created...

**Comment by SimunKaracic** [comment]
We're looking at bringing streams into the fold (no pun intended), but there's a lot of different implementations of them right now, and we'd like some uniform approach to instrumenting all of them.
It's really just a case of trying them all out, and working out what kinds of operation names, tags, metrics, etc. we want to expose from Streams. 

Unfortunately, doing this thing right will take some time :D
